### DISCUSSION TITLE
Need help with pdf coords to viewport point conversion

### DISCUSSION THREAD

**Discussion by jorgebenaventee** [mozilla/pdf.js#19170] [Q&A] [answered]
I am building some application where I need to draw some HTML elements above a pdf. I have a pdf document with a grid and the coordinates of the pdf. I am storing pdf coordinates in each object X and Y property. However, I am getting some offset when positioning my boxes. Here are some snippets.

Box object: 
```javascript
{
  height: 50,
  width: 50,
  x: 500,
  y: 100,
}
```

Box rendering logic:
```javascript
const [viewportX, viewportY] = viewport.convertToViewportPoint(
  box.x,
  box.y,
)

console.log({
  viewportX,
  viewportY,
  boxX: box.x,
  boxY: box.y,
  transform: viewport.transform,
})


container.style.position = 'absolute'
container.style.left = `${viewportX}px`
container.style.top = `${viewportY}px`
container.style.width = `${box.width * viewport.scale}px`
container.style.height = `${box.height * viewport.scale}px`

```

Box position:

![image](https://github.com/user-attachments/assets/5bc6bb80-5b0e-4029-9a59-1ed8587ffdda)

What I am expecting is my box getting correctly positioned in its coords.

Thanks in advance!

**Comment by calixteman** [comment, marked as answer]
You can do something similar to what we do when rendering an annotation:
https://github.com/mozilla/pdf.js/blob/f180de41f3d6f2ad776d4818079db7d2412e5c33/src/display/annotation_layer.js#L265
`data.rect` contains the rect dimensions of the annotation bbox as defined in the pdf.

**Reply by jorgebenaventee** [reply]
It worked! Thank you so much!
